### ISSUE TITLE
Could not initialize class JavadocLexer

### ISSUE THREAD

**Issue by PatrickDattilio** [jshiell/checkstyle-idea#352]
Android Studio 3.0 and AS 3.0 RC 1
Gradle 4.1
Checkstyle 8.2
checkstyle-idea 5.10.2

Running my local checkstyle.xml file fails silently, telling me there are no errors. The same thing happens with the included Google checks, but the sun checks seem to run?

The same error seems to be sent in the logs:

```
java.lang.Error: Error was thrown while processing /var/folders/44/chkbvfdd43564_yshb78k1680000gq/T/csi-159/app/src/main/java/com/MyClass.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:299)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.processAndAudit(OpScan.java:75)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:52)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:22)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:126)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.scan(CheckstyleActionsImpl.java:72)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:45)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$1(CheckStyleInspection.java:91)
	at java.util.Optional.map(Optional.java:215)
	at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:91)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$0(CheckStyleInspection.java:63)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.call(ApplicationImpl.java:357)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocLexer
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.parseJavadocAsParseTree(JavadocDetailNodeParser.java:170)
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.parseJavadocAsDetailNode(JavadocDetailNodeParser.java:122)
	at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.visitToken(AbstractJavadocCheck.java:264)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:425)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:559)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:365)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:210)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:79)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:316)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:286)
	... 16 more
```

**Comment by PatrickDattilio** [comment]
Note that running via command line works just fine (either using the gradle task or a checkstyle jar)

**Comment by jshiell** [comment]
Thanks for the report.

I fear I can't reproduce this with the included Google checks for Checkstyle 8.2. I am using IDEA 2017.2.5, but I don't believe this should affect this (as there's no reason the class path should differ).

Could you please attach your `checkstyle.xml` file? Also, do you have any custom checks installed? Thank you.

**Comment by PatrickDattilio** [comment]
No custom checks installed
checkstyle.xml:
```
<?xml version="1.0"?><!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_3.dtd"><!--
    Checkstyle configuration that checks the Google coding conventions from:
    -  Google Java Style
       https://google-styleguide.googlecode.com/svn-history/r130/trunk/javaguide.html
       https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml
    -  Major league baseball Advance Media
       https://wiki.mlbam.com/display/mobile/Android+Standards

    Checkstyle is very configurable. Be sure to read the documentation at
    http://checkstyle.sf.net (or in your downloaded distribution).
    Most Checks are configurable, be sure to consult the documentation.
    To completely disable a check, just comment it out or delete it from the file.
 -->

<module name="Checker">

    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>
    <property name="fileExtensions" value="java, properties, xml"/>

    <!-- Allow supressing of warnings                           -->
    <!-- http://checkstyle.sourceforge.net/config.html          -->
    <module name="SuppressWarningsFilter"/>
    <module name="SuppressionFilter">
        <property name="file" value="config/checkstyle/checkstyle-suppressions.xml"/>
    </module>

    <!-- Checks that a package-info.java file exists for each package.   -->
    <!-- See http://checkstyle.sf.net/config_javadoc.html#JavadocPackage -->
    <module name="JavadocPackage">
        <property name="severity" value="warning"/>
    </module>

    <!-- Checks whether files end with a new line.                        -->
    <!-- See http://checkstyle.sf.net/config_misc.html#NewlineAtEndOfFile -->
    <module name="NewlineAtEndOfFile">
        <property name="lineSeparator" value="lf" />
    </module>

    <!-- Checks for Size Violations.                    -->
    <!-- See http://checkstyle.sf.net/config_sizes.html -->
    <module name="FileLength"/>

    <!-- Checks for whitespace and report each line          -->
    <!-- See http://checkstyle.sf.net/config_whitespace.html -->
    <module name="FileTabCharacter">
        <property name="eachLine" value="true"/>
    </module>

    <!-- Miscellaneous other checks.                   -->
    <!-- See http://checkstyle.sf.net/config_misc.html -->
    <module name="RegexpSingleline">
        <property name="format" value="\s+$"/>
        <property name="minimum" value="0"/>
        <property name="maximum" value="0"/>
        <property name="message" value="Line has trailing spaces."/>
    </module>

    <!--Only show errors-->
    <!--http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/filters/SeverityMatchFilter.html-->
    <module name="SeverityMatchFilter">
        <property name="severity" value="warning"/>
        <property name="acceptOnMatch" value="false"/>
    </module>

    <module name="TreeWalker">
        <module name="SuppressionCommentFilter"/>

        <!-- Checks for Naming Conventions.                  -->
        <!-- See http://checkstyle.sf.net/config_naming.html -->
        <module name="ConstantName">
            <property name="format" value="^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
        </module>
        <module name="LocalFinalVariableName"/>
        <module name="LocalVariableName">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <property name="allowOneCharVarInForLoop" value="true"/>
            <message key="name.invalidPattern"
                     value="Local variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MemberName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Member name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>
            <message key="name.invalidPattern"
                     value="Method name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="PackageName">
            <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>
            <message key="name.invalidPattern"
                     value="Package name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ParameterName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="StaticVariableName"/>
        <module name="TypeName">
            <message key="name.invalidPattern"
                     value="Type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ClassTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Class type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Method type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="AbbreviationAsWordInName">
            <property name="ignoreFinal" value="false"/>
            <property name="allowedAbbreviationLength" value="1"/>
        </module>

        <!-- Checks for imports                              -->
        <!-- See http://checkstyle.sf.net/config_import.html -->
        <module name="AvoidStarImport"/>
        <module name="IllegalImport"/> <!-- defaults to sun.* packages -->
        <module name="RedundantImport">
            <property name="severity" value="error"/>
        </module>
        <module name="UnusedImports">
            <property name="severity" value="error"/>
            <property name="processJavadoc" value="true"/>
        </module>
        <!-- Android Studio default code formatting requires android at the top,
        followed by com,junit,net,org, java,javax, all other imports, finally all static imports.-->
        <!--http://checkstyle.sourceforge.net/config_imports.html#CustomImportOrder_Example-->
        <module name="CustomImportOrder">
            <property name="customImportOrderRules"
                      value="SPECIAL_IMPORTS###STANDARD_JAVA_PACKAGE###THIRD_PARTY_PACKAGE###STATIC"/>
            <property name="specialImportsRegExp" value="^android|com|junit|net|org|dagger\."/>
            <property name="sortImportsInGroupAlphabetically" value="true"/>
            <property name="separateLineBetweenGroups" value="true"/>
        </module>


        <!-- Checks for Size Violations.                    -->
        <!-- See http://checkstyle.sf.net/config_sizes.html -->
        <module name="LineLength">
            <property name="severity" value="error"/>
            <property name="max" value="120"/>
            <property name="ignorePattern"
                      value="^package.*|^import.*|a href|href|http://|https://|ftp://| * \*.*|static final"/>
        </module>
        <module name="MethodLength"/>
        <module name="ParameterNumber"/>

        <!-- Checks for whitespace                               -->
        <!-- See http://checkstyle.sf.net/config_whitespace.html -->
        <module name="EmptyForIteratorPad"/>
        <module name="GenericWhitespace">
            <message key="ws.followed"
                     value="GenericWhitespace ''{0}'' is followed by whitespace."/>
            <message key="ws.preceded"
                     value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
            <message key="ws.illegalFollow"
                     value="GenericWhitespace ''{0}'' should followed by whitespace."/>
            <message key="ws.notPreceded"
                     value="GenericWhitespace ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="MethodParamPad"/>
        <module name="NoWhitespaceAfter"/>
        <module name="NoWhitespaceBefore"/>
        <module name="OperatorWrap">
            <property name="option" value="NL"/>
            <property name="tokens"
                      value="BAND, BOR, BSR, BXOR, DIV, EQUAL, GE, GT, LAND, LE, LITERAL_INSTANCEOF, LOR, LT, MINUS, MOD, NOT_EQUAL, PLUS, QUESTION, SL, SR, STAR "/>
        </module>
        <module name="ParenPad"/>
        <module name="TypecastParenPad"/>
        <module name="WhitespaceAfter"/>
        <module name="WhitespaceAround">
            <!--<property name="allowEmptyConstructors" value="true"/>-->
            <!--<property name="allowEmptyMethods" value="true"/>-->
            <!--<property name="allowEmptyTypes" value="true"/>-->
            <!--<property name="allowEmptyLoops" value="true"/>-->
            <message key="ws.notFollowed"
                     value="WhitespaceAround: ''{0}'' is not followed by whitespace. Empty blocks may only be represented as '{}' when not part of a multi-block statement (4.1.3)"/>
            <message key="ws.notPreceded"
                     value="WhitespaceAround: ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="NoLineWrap"/>
        <module name="EmptyLineSeparator">
            <property name="allowNoEmptyLineBetweenFields" value="true"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="DOT"/>
            <property name="option" value="nl"/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="COMMA"/>
            <property name="option" value="EOL"/>
        </module>

        <!-- Modifier Checks                                    -->
        <!-- See http://checkstyle.sf.net/config_modifiers.html -->
        <module name="ModifierOrder"/>
        <module name="RedundantModifier"/>

        <!-- Checks for blocks. You know, those {}'s         -->
        <!-- See http://checkstyle.sf.net/config_blocks.html -->
        <module name="AvoidNestedBlocks"/>
        <module name="EmptyBlock">
            <property name="option" value="TEXT"/>
            <property name="tokens"
                      value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH, LITERAL_CATCH"/>
        </module>
        <module name="LineLength">
            <property name="max" value="120"/>
        </module>
        <module name="NeedBraces"/>
        <module name="RightCurly">
            <property name="option" value="alone"/>
            <property name="tokens"
                      value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, LITERAL_DO, STATIC_INIT, INSTANCE_INIT"/>
        </module>
        <module name="EmptyCatchBlock">
            <property name="exceptionVariableName" value="expected"/>
        </module>

        <!-- Checks for common coding problems               -->
        <!-- See http://checkstyle.sf.net/config_coding.html -->
        <module name="EmptyStatement"/>
        <module name="EqualsHashCode"/>
        <module name="IllegalInstantiation"/>
        <module name="InnerAssignment"/>
        <module name="MagicNumber">
            <property name="ignoreHashCodeMethod" value="true" />
        </module>
        <module name="MissingSwitchDefault"/>
        <module name="SimplifyBooleanExpression"/>
        <module name="SimplifyBooleanReturn"/>
        <module name="IllegalTokenText">
            <property name="tokens" value="STRING_LITERAL, CHAR_LITERAL"/>
            <property name="format"
                      value="\\u00(08|09|0(a|A)|0(c|C)|0(d|D)|22|27|5(C|c))|\\(0(10|11|12|14|15|42|47)|134)"/>
            <property name="message" value="Avoid using corresponding octal or Unicode escape."/>
        </module>
        <module name="OneStatementPerLine"/>
        <module name="MultipleVariableDeclarations"/>
        <module name="FallThrough"/>
        <module name="NoFinalizer"/>
        <module name="OverloadMethodsDeclarationOrder"/>
        <module name="VariableDeclarationUsageDistance"/>

        <!-- Checks for class design                         -->
        <!-- See http://checkstyle.sf.net/config_design.html -->
        <module name="DesignForExtension">
            <property name="severity" value="warning"/>
        </module>

        <module name="HideUtilityClassConstructor"/>
        <module name="InterfaceIsType"/>
        <module name="VisibilityModifier">
            <property name="ignoreAnnotationCanonicalNames"
                      value="android.support.annotation.VisibleForTesting, org.junit.Rule, javax.inject.Inject, Bind,
                      BindView, BindString, BindInt, BindBool, BindColor, BindDimen, BindDrawable, BindViews, Mock,
                      Captor, EpoxyAttribute"
            />
        </module>
        <module name="OneTopLevelClass"/>

        <!-- Miscellaneous other checks.                   -->
        <!-- See http://checkstyle.sf.net/config_misc.html -->
        <module name="ArrayTypeStyle"/>
        <module name="FinalParameters">
            <property name="severity" value="warning"/>
        </module>
        <module name="TodoComment">
            <property name="format" value="ERROR"/>
        </module>
        <module name="UpperEll"/>
        <module name="OuterTypeFilename"/>
        <module name="AvoidEscapedUnicodeCharacters">
            <property name="allowEscapesForControlCharacters" value="true"/>
            <property name="allowByTailComment" value="true"/>
            <property name="allowNonPrintableEscapes" value="true"/>
        </module>
        <module name="UpperEll"/>
        <module name="Indentation">
            <!--This is set to a warning until https://github.com/checkstyle/checkstyle/issues/3182 is resolved. -->
            <property name="severity" value="warning"/>
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
            <property name="tabWidth" value="4"/>
        </module>

        <!-- Checks for style with the usage of annotations               -->
        <!-- See http://checkstyle.sourceforge.net/config_annotation.html -->
        <module name="AnnotationLocation">
            <property name="tokens"
                      value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF"/>
        </module>
        <module name="AnnotationLocation">
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="allowSamelineMultipleAnnotations" value="true"/>
        </module>
        <module name="MissingOverride"/>

        <!-- Checks for Javadoc comments.                     -->
        <!-- See http://checkstyle.sf.net/config_javadoc.html -->
        <module name="JavadocMethod">
            <property name="severity" value="error"/>
            <property name="scope" value="public"/>
            <property name="minLineCount" value="2"/>
            <property name="allowedAnnotations"
                      value="Override, Test, Singleton, Provides, OnTouch, Before"/>
            <property name="allowThrowsTagsForSubclasses" value="true"/>
            <property name="allowMissingPropertyJavadoc" value="true"/>
        </module>
        <module name="JavadocType"/>
        <module name="JavadocStyle"/>
        <module name="NonEmptyAtclauseDescription"/>
        <module name="JavadocTagContinuationIndentation"/>
        <module name="SummaryJavadocCheck">
            <property name="forbiddenSummaryFragments"
                      value="^@return the *|^This method returns |^A [{]@code [a-zA-Z0-9]+[}]( is a )"/>
        </module>
        <module name="JavadocParagraph"/>
        <module name="AtclauseOrder">
            <property name="tagOrder" value="@param, @return, @throws, @deprecated"/>
            <property name="target"
                      value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
        </module>

        <!-- Allow suppression of warnings for tree                 -->
        <!-- http://checkstyle.sourceforge.net/config.html          -->
        <module name="SuppressWarningsHolder"/>
    </module>
</module>
```

**Comment by jshiell** [comment]
Thanks. The bad news is that that works perfectly for me in the plugin. 

I've double-checked and `JavadocLexer` is part of the core Checkstyle JAR, so if the rest of Checkstyle is available then this should also be. 

Which leads me to voodoo, as I've no better suggestions - could you please try deleting the plugin entirely and pull it down again, just in case the JAR is corrupt? It's madness, but I'm short on ideas given that `TreeWalker`, from the same JAR, is very clearly visible in the stack trace.

**Comment by dk-tkennedy** [comment]
Also having this same issue on Android Studio 3.0. Tried uninstalling and reinstalling the plug in and that didn't fix it for me.

**Comment by jshiell** [comment]
Thanks for the feedback. Does it work okay for you in IDEA? Android Studio support has always been on a fingers-crossed basis, but it’d be odd if the products had diverged enough to cause this sort of oddity,

**Comment by PatrickDattilio** [comment]
Verified that checkstyle-idea 5.11 works in IDEA CE 2017.2.5 , but not Android Studio 3.0. In AS it now tells me to check the Event Log (but the error is not shown in the event log). In the log file, I get the same error:

```
2017-11-01 10:16:05,452 [d thread 6]   WARN - heckstyle.CheckStyleInspection - The inspection could not be executed. 
java.lang.Error: Error was thrown while processing /Users/patrickdattilio/dev/MyProject/app/src/main/java/com/example/MyFile.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:299)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.processAndAudit(OpScan.java:75)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:52)
	at org.infernus.idea.checkstyle.service.cmd.OpScan.execute(OpScan.java:22)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.executeCommand(CheckstyleActionsImpl.java:126)
	at org.infernus.idea.checkstyle.service.CheckstyleActionsImpl.scan(CheckstyleActionsImpl.java:72)
	at org.infernus.idea.checkstyle.checker.CheckStyleChecker.scan(CheckStyleChecker.java:45)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$inspectFile$1(CheckStyleInspection.java:91)
	at java.util.Optional.map(Optional.java:215)
	at org.infernus.idea.checkstyle.CheckStyleInspection.inspectFile(CheckStyleInspection.java:91)
	at org.infernus.idea.checkstyle.CheckStyleInspection.lambda$checkFile$0(CheckStyleInspection.java:63)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.call(ApplicationImpl.java:357)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocLexer
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.parseJavadocAsParseTree(JavadocDetailNodeParser.java:170)
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.parseJavadocAsDetailNode(JavadocDetailNodeParser.java:122)
	at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.visitToken(AbstractJavadocCheck.java:264)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:425)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:559)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:365)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:210)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:79)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:316)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:286)
	... 16 more
```

**Comment by beworker** [comment]
After upgrading to Android Studio 3.0.0 I am experiencing the same issue. The error only happens when some javadoc rules are active.

```
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocLexer
```

This error rather means that `JavadocLexer` tries to import some classes, which are missing. That's why `JavadocLexer` cannot be initialized. It might be, that Android Studio does not pack some framework classes, which are present in original IDEA package. Can this somehow be helpful?

**Comment by jshiell** [comment]
The honest answer is I have no idea what is going on here, or even how Android Studio manages to break this. We build a classpath with the selected JAR, so you should either get all the Checkstyle classes or - if it breaks - nothing. Yet in this case we seem to have all of them expect `JavadocLexer`. Which is a bit messy, as JavadocLexer is packaged as part of the core Checkstyle JAR, e.g.
```
$ find ~/Library/Application\ Support/IntelliJIdea2017.2/CheckStyle-IDEA/checkstyle/lib -name 'checkstyle-*' -exec unzip -l \{\} \; | grep -e JavadocLexer -e jar
Archive:  /Users/jshiell/Library/Application Support/IntelliJIdea2017.2/CheckStyle-IDEA/checkstyle/lib/checkstyle-7.1.1.jar
    14716  09-05-2016 07:08   com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4
    49202  09-05-2016 07:08   com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.class
Archive:  /Users/jshiell/Library/Application Support/IntelliJIdea2017.2/CheckStyle-IDEA/checkstyle/lib/checkstyle-6.13.jar
    49213  11-30-2015 06:37   com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.class
    14715  11-30-2015 06:37   com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocLexer.g4
(and so on)
```
The class in question isn't even doing anything clever to instantiate the lexer, just a vanilla `new`:
```
final JavadocLexer lexer = new JavadocLexer(input);
```
TL;DR so far no progress whatsoever.

**Comment by beworker** [comment]
I believe I was able to enable more traces in Android Studio and here is the error thrown.

```
Caused by: java.lang.NoSuchMethodError: org.antlr.v4.runtime.atn.RuleTransition.<init>(Lorg/antlr/v4/runtime/atn/RuleStartState;IILorg/antlr/v4/runtime/atn/ATNState;)V
	at org.antlr.v4.runtime.atn.ATNDeserializer.edgeFactory(ATNDeserializer.java:715)
	at org.antlr.v4.runtime.atn.ATNDeserializer.deserialize(ATNDeserializer.java:332)
	at com.puppycrawl.tools.checkstyle.grammars.javadoc.JavadocLexer.<clinit>(JavadocLexer.java:2053)
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.createJavadocParser(JavadocDetailNodeParser.java:174)
	at com.puppycrawl.tools.checkstyle.JavadocDetailNodeParser.parseJavadocAsDetailNode(JavadocDetailNodeParser.java:123)
	at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.visitToken(AbstractJavadocCheck.java:297)
	at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:423)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:579)
	at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:363)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:208)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:82)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:318)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:288)
	... 18 more
```

As I was mentioning before the error happens not because of missing lexer class, but because lexer class requires something what is missing. Does `NoSuchMethodError` above means anything to you?

**Comment by jshiell** [comment]
Thanks, I misread your previous comment. That stacktrace suggests that it's not using the correct version of Antlr for the selected version of Checkstyle. 

Does this only happen with Checkstyle 8.2 selected or with all versions?

**Comment by beworker** [comment]
Just tried with 8.0 to 8.4 versions. Even restarted IDE after changing the version. All of them fail.   Standard Google Checks were used.

**Comment by jshiell** [comment]
I have a theory. No more than that, and it's unlikely to get further today as I'm full of flu, but it's suggestive.

The classpath does, of course, inherit from the existing IDEA classpath. This means it's possible to pollute the classpath above. Checkstyle 8.2 uses Antlr 4.7 - however, a quick poke around IDEA suggests that the Android plugin uses 4.5.3, i.e.
```
find /Applications/IntelliJ\ IDEA.app -name '*antlr*'                          jshiell@LimitingFactor
/Applications/IntelliJ IDEA.app/Contents/plugins/android/lib/antlr4-runtime-4.5.3.jar
/Applications/IntelliJ IDEA.app/Contents/plugins/svn4idea/lib/antlr.jar
/Applications/IntelliJ IDEA.app/Contents/lib/ant/lib/ant-antlr.jar
/Applications/IntelliJ IDEA.app/Contents/lib/ant/lib/ant-antlr.pom
```
This sits in a plugin in IDEA, hence the classpath should be isolated - and, given we don't see this on IDEA, this seems to hold true. However - and this is where I drop into conjecture - this may not be true in Android Studio, given it's a specialised version of the IDEA platform. Hence, this may be available on the parent classpath and it's possible that the plugin is hence seeing a mix of versions (4.7 at the local level, 4.5.3 on the parent) and hence getting into a kerfuffle.

As a fix, I theorise we'd have to modify the local class loader to explicitly block lookups on the Antlr packages from being delegated to the parent classloader (e.g. anything in `org.antlr`). Which gives us a vector of attack, at least.

**Comment by jshiell** [comment]
Done and released as 5.12.1. Seems to resolve the problem with local testing. Please shout and reopen if you do not see similar results.

Thanks very much for your help in resolving this.
